### ISSUE TITLE
Broadcast of Intent { act=android.intent.action.SCREEN_ON flg=0x50200010 }

### ISSUE THREAD

**Issue by Javsons** [libgdx/libgdx#6051]
Android Live Wallpaper

` 
  at java.lang.Object.wait (Object.java)
- waiting on <0x01f37d2d> (a java.lang.Object)
 
  at com.badlogic.gdx.backends.android.AndroidGraphicsLiveWallpaper.resume (AndroidGraphicsLiveWallpaper.java)
- locked <0x01f37d2d> (a java.lang.Object)
 
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaper.onResume (AndroidLiveWallpaper.java)
 
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onResume (AndroidLiveWallpaperService.java)
 
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.notifyVisibilityChanged (AndroidLiveWallpaperService.java)
 
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onVisibilityChanged (AndroidLiveWallpaperService.java)
 
  at android.service.wallpaper.WallpaperService$Engine.reportVisibility (WallpaperService.java:1088)
 
  at android.service.wallpaper.WallpaperService$Engine$3.onDisplayChanged (WallpaperService.java:1224)
 
  at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleMessage (DisplayManagerGlobal.java:633)
 
  at android.os.Handler.dispatchMessage (Handler.java:106)
 
  at android.os.Looper.loop (Looper.java:193)
 
  at android.app.ActivityThread.main (ActivityThread.java:6718)
 
  at java.lang.reflect.Method.invoke (Method.java)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)`

`"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x74f20910 self=0x7288a14c00
  | sysTid=25299 nice=0 cgrp=default sched=0/0 handle=0x730e404548
  | state=S schedstat=( 103212346 86202757 247 ) utm=5 stm=3 core=0 HZ=100
  | stack=0x7fe8581000-0x7fe8583000 stackSize=8MB
  | held mutexes=`

any 1 know what to do with that ANR?

**Comment by oshetima** [comment]
From my experience, I can say that this is a bug mainly on Android 9 phones and I'm confident enough that it has nothing to do with libgdx. Probably it will never be solved. 
You can reproduce it by following these steps:
1. Take an Android 9.x phone or emulator
2. Go to the live wallpaper preview screen
3. Press recent apps button 
4. From recent apps choose your live wallpaper preview

The screen will go black with an ANR. I have thousands of these every day and thanks God I'm just below the bad behavior threshold limit. This ANR is thrown by any live wallpaper (even the prebuild ones) on Android 9 so there is no hope.

**Comment by SimonIT** [comment]
I tried it it with the included LiveWallpaper test and can reproduce a black screen after doing your steps. 
I looked inside the logcat and found these two errors:
```
2020-05-18 19:36:48.904 8424-8450/com.badlogic.gdx.tests.android E/libEGL: eglCreateWindowSurface: native_window_api_connect (win=0xec336008) failed (0xffffffed) (already connected to another API?)
2020-05-18 19:36:48.904 8424-8450/com.badlogic.gdx.tests.android E/libEGL: eglCreateWindowSurface:720 error 3003 (EGL_BAD_ALLOC)
```
I'm not sure if they are related, but I thought posting them here is maybe useful.
By doing some research, I found [this](https://stackoverflow.com/questions/54183090/eglcreatewindowsurface-native-window-api-connect-failed) question on stackoverflow. I wasn't able to fix the problem but maybe someone with more knowledge can

**Comment by oshetima** [comment]
Yes @SimonIT, this is exactly what happens. I've already reported it to Google IssueTracker for Android [here](https://issuetracker.google.com/issues/142331942) in October 2019 and they initially accepted the bug but now it is marked as "Won't Fix (Obsolete)". I also reported it here #5723, so I presume there is no hope...

**Comment by SimonIT** [comment]
I tried it again and I recognized that somehow the mentioned `onWindowFocusChanged` method never gets called.
I really don't know why. If you wanna try it yourself, I uploaded it to a seperate branch of my fork https://github.com/SimonIT/libgdx/tree/fix-Context-Loss-Live-Wallpaper

**Comment by Javsons** [comment]
@SimonIT 
Thanks for that help... now i know where its problem.

@oshetima 
yeah its on andorid 9.0, but others apps dont give ANRs, just show "We got problem" and black screen
